pinot-common/src/main/java/org/apache/pinot/common/utils/helix/LeadControllerUtils.java

```diff
@@ -58,4 +68,37 @@ public static String generatePartitionName(int partitionId) {
   public static int extractPartitionId(String partitionName) {
     return Integer.parseInt(partitionName.substring(partitionName.lastIndexOf('_') + 1));
   }
+
+  /**
+   * Checks from ZK if resource config of leadControllerResource is enabled.
+   * @param helixManager helix manager
+   */
+  public static boolean isLeadControllerResourceEnabled(HelixManager helixManager) {
+    ConfigAccessor configAccessor = helixManager.getConfigAccessor();
+    ResourceConfig resourceConfig =
+        configAccessor.getResourceConfig(helixManager.getClusterName(), Helix.LEAD_CONTROLLER_RESOURCE_NAME);
+    String resourceEnabled = resourceConfig.getSimpleConfig(Helix.LEAD_CONTROLLER_RESOURCE_ENABLED_KEY);
+    return Boolean.parseBoolean(resourceEnabled);
+  }
+
+  /**
+   * Gets Helix leader in the cluster. Null if there is no leader.
+   * @param helixManager helix manager
+   * @return instance id of Helix cluster leader, e.g. localhost_9000.
+   */
+  public static String getHelixClusterLeader(HelixManager helixManager) {
```

Comment:
Why is this method here and not in ControllerLeaderLocator?

Reply:
This method can be called by `LeadControllerManager` from Pinot controller to check whether the current controller is helix leader as well. So it's needed to be here for both controller and server.